src/cockpit/channels/stream.py

```diff
@@ -75,7 +76,11 @@ def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, A
             stderr = subprocess.PIPE
 
         env: Dict[str, str] = dict(os.environ)
-        env.update(options.get('env') or [])
+        if environ:
+            if not isinstance(environ, list):
+                raise ChannelError('protocol-error', message='invalid "environ" option for stream channel')
```

Comment:
But if you want to do that, then you also need to check if (a) every item is a string, and (b) every item contains an `=` sign.

Reply:
I suppose, the C bridge does `cockpit_json_get_strv` which does:
```
 * Gets a string array member from a JSON object. Validates
 * that the member is an array, and that all elements in the
 * array are strings. If these fail, then will return %FALSE.
```